### PR TITLE
tests; bug fixes; docs

### PR DESCRIPTION
**Pull request by Fenix-125**
- remake the test colling function;
- add header and the skreatch for 'test_my_str_find';
- made some fixed or modifications in functions:
            my_str_find, my_str_cmp, my_str_cmp_cstr, my_str_find_c,
            my_str_find_if, my_str_write_file, my_str_read_file_delim, char_arr_len;
- write documentation for some functions issue #15 ;
- write tests issue #16 :
            test_my_str_read_file_delim, test_my_str_write_file, test_my_str_read_file,
            test_my_str_find_if, test_my_str_find_c, test_my_str_cmp_cstr, test_my_str_cmp;

### CODE REVIEW

File: lib/src/c_string.c [hunk 1]
```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
...
 }
 
-// TODO: test my_str_cmp_cstr
 int my_str_cmp_cstr(const my_str_t *str1, const char *cstr2) {
-    // compare string with cstring
-    // behavior is the same as in strcmp
-    // return 0 if there are equal, -1 if first is less, 1 if second is less
-    size_t str2_len = char_arr_len(cstr2);
-    if (str1->size_m > str2_len)
+    // This function return values that are as follows:
```
Comment (Myralllka) [inline lib/src/c_string.c]: Please leave my documentation :)

File: lib/src/c_string.c [hunk 2]
```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
...
     return (size_t) (-1);
 }
 
-// TODO: test my_str_find_if
 size_t my_str_find_if(const my_str_t *str, int (*predicat)(int)) {
-    // find symbol in the string for whish function returns true
-    // return it`s index or (size_t) (-1) if there no such character
-    for (int j = 0; j < str->size_m; ++j) {
-        if (predicat(str->data[j]))
+    // Find the first occurrence of a char that satisfy the 'predicat' function
```
Comment (Myralllka) [inline lib/src/c_string.c]: This line is better than my

File: lib/src/c_string.c [hunk 3]
```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
...
+    // If there is not such char in the string returns -1
```
Comment (Myralllka) [inline lib/src/c_string.c]: Not this, I think
It's better to write what function returns in one, last line

File: lib/src/c_string.c [hunk 4]
```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
...
             return j;
     }
     return (size_t) (-1);
 }
 
 int my_str_read_file(my_str_t *str, FILE *file) {
-    // read string from file
-    // it reads all file
-    // return 0 if everything ok, negative numbers otherwise
+    // read the file to the EOF
```
Comment (Myralllka) [inline lib/src/c_string.c]: This line is better :) return in one line, please

File: lib/src/c_string.c [hunk 5]
```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
...
+    // return -1 if can not add a character from the file to the end of the string
+    // return 0 if everything is Ok
     return my_str_read_file_delim(str, file, EOF);
 }
 
 int my_str_read(my_str_t *str) {
-    //analog of my_str_read_file but from stdin
+    // read the input from the standard input till the EOF
+    // return -1 if can not add a character from the file to the end of the string
```
Comment (Myralllka) [inline lib/src/c_string.c]: I'm not sure that we need this documentation as a copy of previous

File: lib/src/c_string.c (hunk 1 above)
Comment (Fenix-125) [reply lib/src/c_string.c]: Ok

File: lib/src/c_string.c (hunk 5 above)
Comment (Fenix-125) [reply lib/src/c_string.c]: OK

File: lib/src/c_string.c (hunk 3 above)
Comment (Fenix-125) [reply lib/src/c_string.c]: OK

**Pull request merged**